service/resource/kmskeyv2/delete.go

```diff
@@ -0,0 +1,78 @@
+package kmskeyv2
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/kms"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/framework"
+)
+
+func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange interface{}) error {
+	deleteAliasInput, err := toDeleteAliasInput(deleteChange)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	keyAlias := deleteAliasInput.AliasName
+	if keyAlias != nil && *keyAlias != "" {
+		if _, err := r.awsClients.KMS.DeleteAlias(&deleteAliasInput); err != nil {
+			return microerror.Mask(err)
+		}
+
+		key, err := r.awsClients.KMS.DescribeKey(&kms.DescribeKeyInput{
+			KeyId: keyAlias,
+		})
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		// AWS API doesn't allow to delete the KMS key immediately, but we can schedule its deletion
```

Comment:
Nitpick but can you make the comment end with a full stop.

Reply:
Absolutely :) will try to not forget